spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/VaultEnvironmentRepositoryFactory.java

```diff
@@ -35,8 +45,28 @@ public VaultEnvironmentRepositoryFactory(ObjectProvider<HttpServletRequest> requ
 
 	@Override
 	public VaultEnvironmentRepository build(VaultEnvironmentProperties environmentProperties) {
-		VaultEnvironmentRepository repository = new VaultEnvironmentRepository(request, watch, new RestTemplate(),
+		RestTemplate restTemplate = environmentProperties.isSkipSslValidation() ? skipSslValidationRestTemplate() :
+				new RestTemplate();
+		VaultEnvironmentRepository repository = new VaultEnvironmentRepository(request, watch, restTemplate,
 				environmentProperties);
 		return repository;
 	}
+
+	private RestTemplate skipSslValidationRestTemplate() {
+		try {
+			SSLContext sslContext = new SSLContextBuilder()
+					.loadTrustMaterial(null, (certificate, authType) -> true)
+					.build();
+			CloseableHttpClient httpClient
+					= HttpClients.custom()
+					.setSSLContext(sslContext)
+					.setSSLHostnameVerifier(new NoopHostnameVerifier())
+					.build();
+			HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
+			requestFactory.setHttpClient(httpClient);
+			return new RestTemplate(requestFactory);
```

Comment:
It would be useful (especially with SCS proxy support) to specify additional `HttpClientFactory` settings.